django/template/defaultfilters.py

```diff
@@ -57,6 +58,26 @@ def _dec(*args, **kwargs):
 # STRINGS         #
 ###################
 
+@stringfilter
+def autoescape_aware(func):
+    """
+    Decorator for filters which should automatically escape text based
+    on whether or not autoescape is True.
+    """
+    def _dec(*args, **kwargs):
+        try:
+            autoescape = kwargs['autoescape']
+        except KeyError:
+            autoescape = signature(func).parameters['autoescape'].default
```

Comment:
A `try ... except` block is extremely efficient if no exceptions are raised but catching an exception is expensive. Could you please remove the above code and refactor it to the following:

```python
if 'autoescape' in kwargs:
  autoescape = kwargs['autoescape']
else:
  autoescape = signature(func).parameters['autoescape'].default
```